tests/manage/test_pvc_disruptive.py

```diff
@@ -21,37 +31,174 @@ class BaseDisruption(ManageTest):
+        return pvc_objs
+
     def disruptive_base(self, operation_to_disrupt, resource_to_delete):
         """
-        Base function for PVC disruptive tests
+        Base function for PVC disruptive tests.
+        Deletion of 'resource_to_delete' will be introduced while
+        'operation_to_disrupt' is progressing.
         """
-        DISRUPTION_OPS.set_resource(resource=resource_to_delete)
-        self.pvc_obj = helpers.create_pvc(
-            sc_name=self.sc_obj.name, wait=False
+        # Fetch the number of Pods and PVCs
+        initial_num_of_pods = len(pod.get_all_pods(namespace=self.namespace))
+        initial_num_of_pvc = len(
+            get_all_pvcs(namespace=self.namespace)['items']
         )
+
+        executor = ThreadPoolExecutor(max_workers=1)
+
+        DISRUPTION_OPS.set_resource(resource=resource_to_delete)
+
+        # Start creation of multiple PVCs
+        bulk_pvc_create = executor.submit(self.create_pvcs)
+
         if operation_to_disrupt == 'create_pvc':
-            DISRUPTION_OPS.delete_resource()
-        self.pvc_obj.reload()
-        assert self.pvc_obj.ocp.wait_for_resource(
-            condition=constants.STATUS_BOUND, resource_name=self.pvc_obj.name, timeout=120
-        )
+            # Ensure PVCs are being created before deleting the resource
+            for sample in TimeoutSampler(10, 1, get_all_pvcs, self.namespace):
+                current_num_of_pvc = len(sample['items'])
+                if current_num_of_pvc > initial_num_of_pvc:
+                    num_of_new_pvc = current_num_of_pvc - initial_num_of_pvc
+                    logging.info(
+                        f"PVCs creation has started. "
+                        f"Found {num_of_new_pvc} new PVC."
+                    )
+                    DISRUPTION_OPS.delete_resource()
+                    break
+
+        pvc_objs = bulk_pvc_create.result()
 
-        self.pod_obj = helpers.create_pod(
-            interface_type=constants.CEPHBLOCKPOOL, pvc_name=self.pvc_obj.name, wait=False
+        # Verify PVCs are Bound
+        for pvc_obj in pvc_objs:
+            assert pvc_obj.ocp.wait_for_resource(
+                condition=constants.STATUS_BOUND, resource_name=pvc_obj.name,
+                timeout=120
+            ), (
+                f"Wait timeout: PVC {pvc_obj.name} is not in 'Bound' status "
+                f"even after 120 seconds."
+            )
+        logging.info("Verified: PVCs are Bound.")
+
+        # Start creating pods
+        bulk_pod_create = executor.submit(
+            helpers.create_pods, pvc_objs_list=pvc_objs,
+            interface_type=self.interface, wait=False,
+            namespace=self.namespace
         )
+
         if operation_to_disrupt == 'create_pod':
-            DISRUPTION_OPS.delete_resource()
-        self.pod_obj.reload()
+            # Ensure that pods are being created before deleting the resource
+            for sample in TimeoutSampler(
```

Comment:
I suggest you create a function that wraps this so it will be called from here and from line #74

Reply:
Created new function for getting the number of newly added pvc/pod